envoy-control-core/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/groups/NodeMetadata.kt

```diff
@@ -17,6 +17,11 @@ class NodeMetadata(metadata: Struct, properties: SnapshotProperties) {
         .fieldsMap["service_name"]
         ?.stringValue
 
+    val ads = metadata
```

Comment:
It's weird that throughout the whole codebase we use an `ads` bool flag to determine what mode we operate in. Sounds like a perfect use-case for an enum. Would also make tests more readable: now we have indirect notation like `cache.getSnapshot(AllServicesGroup(ads = false))` which means "get snapshot for XDS".

I think it's a good opportunity to make such refactor, if more people agree.

Reply:
Fixed